### PR TITLE
Updates to be C# 5 compatible for MonoGame console use

### PR DESCRIPTION
**Pull request by JakeLegendXIII**
#25 Making the code compatible with C# 5 in order to allow MonoGame to update to the newer version and maintain existing Console support. 

Primarily cleaning up expression-body properties, automatically implemented readonly properties, one usage of nameof was cleaned up, and usage of stackalloc initializers in the generated code.

Please let me know if there are any issues or concerns. I know a PR like this is not ideal, and hopefully we'll get past the C# 5 limitations in MonoGame. The generated code updates were pulled from a previous version.

### CODE REVIEW

File: src/StbImage.Generated.Png.cs [hunk 1]
```diff
@@ -570,7 +598,8 @@ public static int stbi__parse_png_file(stbi__png z, int scan, int req_comp)
 			var palette = stackalloc byte[1024];
 			byte pal_img_n = 0;
 			byte has_trans = 0;
-			var tc = stackalloc byte[] { 0, 0, 0 };
+			var tc = stackalloc byte[3];
+			tc[0] = 0;
```
Comment (rds1983) [inline src/StbImage.Generated.Png.cs:602]: shouldn't other values of that array be initialized to zero too?

Comment (JakeLegendXIII) [reply src/StbImage.Generated.Png.cs:602]: Yeah I can do that if you'd like. This is how it is in the version currently used in MonoGame back on this commit: https://github.com/StbSharp/StbImageSharp/blob/55e833e7f0f86b247040dabfbad51ca46f164118/src/StbImage.Generated.Png.cs#L557

It only had the first initialized in the older version same as raw_data one in Tga below, but it should be fine to initialize them as 0's just like the newer code was doing. I can update both if you'd prefer!

Comment (rds1983) [reply src/StbImage.Generated.Png.cs:602]: yes, please update both.

Comment (JakeLegendXIII) [reply src/StbImage.Generated.Png.cs:602]: Updated.

**Comment by rds1983** [comment]
thanks @JakeLegendXIII!

**Pull request merged**